node.go

```diff
@@ -16,16 +20,20 @@ var (
 	ErrLinkNotFound = fmt.Errorf("no link by that name")
 )
 
+type immutableProtoNode struct {
+	encoded []byte
+	dagpb.PBNode
+}
+
 // ProtoNode represents a node in the IPFS Merkle DAG.
 // nodes have opaque data and a set of navigable links.
 type ProtoNode struct {
-	links []*ipld.Link
+	links []*format.Link
 	data  []byte
 
 	// cache encoded/marshaled value
-	encoded []byte
-
-	cached cid.Cid
+	encoded *immutableProtoNode
+	cached  cid.Cid
```

Comment:
I think I need a comment block on these types that briefly describes what the difference is between these and [go-codec-dagpb.PBNode](https://pkg.go.dev/github.com/ipld/go-codec-dagpb#PBNode).

Maybe it's all legacy, maybe it's actually somewhat functional...?  It seems like it will be hard for a reader to tell without a large amount of splunking.